Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1813,6 +1813,11 @@ static LayoutUnit initialJustifyContentOffset(const RenderStyle& style, LayoutUn
     ContentPosition justifyContent = style.resolvedJustifyContentPosition(contentAlignmentNormalBehavior());
     ContentDistribution justifyContentDistribution = style.resolvedJustifyContentDistribution(contentAlignmentNormalBehavior());
 
+    if (availableFreeSpace < 0 && style.justifyContent().overflow() == OverflowAlignment::Safe) {
```

Comment:
Why is the third condition in a separate `if` statement ?

Reply:
It broke up the logic into the two logical phases: do we need to perform safe alignment, and do we need to change the value of justifyContent to make it happen.

Reply:
Although, I guess technically you can't have normal safe alignment, so maybe i can just make that an assert.